### ISSUE TITLE
[2.9.1] Generic.WhiteSpace.ScopeIndent `if` open scope `{` with too many tabs ignored.

### ISSUE THREAD

**Issue by photodude** [squizlabs/PHP_CodeSniffer#1815]
according to my understanding the `Generic.WhiteSpace.ScopeIndent` sniff for things like `if` open new scopes should be exact indenting. 

I'm seeing no error when there are too many tabs on an `if` scope opener `{`  inside a closure

verbose reports with too many and too few tabs attached.

example code expected no error (or fixed result)
```php
// Sort the array by value of subarray
usort(
	$languages,
	function($a, $b) use ($that)
	{
		$ordering = $that->getState('list.ordering');

		if (strtolower($that->getState('list.direction')) === 'asc')
		{
			return StringHelper::strcmp($a->$ordering, $b->$ordering);
		}
		else
		{
			return StringHelper::strcmp($b->$ordering, $a->$ordering);
		}
	}
);
```

example where Generic.WhiteSpace.ScopeIndent should error
```php
// Sort the array by value of subarray
usort(
	$languages,
	function($a, $b) use ($that)
	{
		$ordering = $that->getState('list.ordering');

		if (strtolower($that->getState('list.direction')) === 'asc')
			{
			return StringHelper::strcmp($a->$ordering, $b->$ordering);
		}
		else
		{
			return StringHelper::strcmp($b->$ordering, $a->$ordering);
		}
	}
);
```

example code where the expected error is correctly found
```php
// Sort the array by value of subarray
usort(
	$languages,
	function($a, $b) use ($that)
	{
		$ordering = $that->getState('list.ordering');

		if (strtolower($that->getState('list.direction')) === 'asc')
	{
			return StringHelper::strcmp($a->$ordering, $b->$ordering);
		}
		else
		{
			return StringHelper::strcmp($b->$ordering, $a->$ordering);
		}
	}
);
```
verbose output where sniff correctly errors with too few tabs on an `if` scope opener `{`  inside a closure
[output-too few tabs.txt](https://github.com/squizlabs/PHP_CodeSniffer/files/1589212/output-too.few.tabs.txt)

verbose output where sniff does not error with too many tabs on an `if` scope opener `{`  inside a closure
[output-too many tabs.txt](https://github.com/squizlabs/PHP_CodeSniffer/files/1589213/output-too.many.tabs.txt)

**Comment by gsherwood** [comment]
This sniff doesn't enforce an exact indent for opening braces. If uses the IF statement itself to set the min indent level for all lines within that block, which is why you get an error when the indent is too small.

There are no included sniffs that enforce brace placement on the line following the statement, as you have in your example code. There is one that bans that syntax, but not one that enforces it, or allows both methods.

**Comment by photodude** [comment]
Did I misunderstand your comment in https://github.com/squizlabs/PHP_CodeSniffer/issues/387#issuecomment-67997324 I thought you were saying there that the indentation for functions and IF statements was exact but otherwise was indented at least to the required level?

**Comment by photodude** [comment]
Our custom [Joomla.ControlStructures.ControlStructuresBrackets](https://github.com/joomla/coding-standards/blob/master/Joomla/Sniffs/ControlStructures/ControlStructuresBracketsSniff.php) sniff based on [PEAR.Classes.ClassDeclaration](https://github.com/squizlabs/PHP_CodeSniffer/blob/2.9/CodeSniffer/Standards/PEAR/Sniffs/Classes/ClassDeclarationSniff.php) Sniff handles the enforcement of the brace placement on the line following the statement.

The issue here is only the indentation.

Our custom sniff is not indenting enough with closures as above, the opening brace of an `if` inside a closure ends up a tab short. The custom sniff section on indenting is based on the `$tokens[$stackPtr]['level']` which reports a level less than expected with the code above inside a closure. but otherwise reports correctly. 

Generic.WhiteSpace.ScopeIndent does a good job getting the indenting right, but in the case of too much indenting it doesn't throw an error as expected.

**Comment by gsherwood** [comment]
> Did I misunderstand your comment in #387 (comment) I thought you were saying there that the indentation for functions and IF statements was exact but otherwise was indented at least to the required level or more?

The indent for the line where the FUNCTION or IF (or whatever) token is is enforced exactly. The brace position has never been enforced in that way though.

**Comment by photodude** [comment]
Thanks for the clarification @gsherwood 
Closing as expected behavior for this sniff.

**Comment by gsherwood** [comment]
> Our custom Joomla.ControlStructures.ControlStructuresBrackets sniff based on PEAR.Classes.ClassDeclaration Sniff handles the enforcement of the brace placement on the line following the statement.

Instead of using the `level` array index, why not just use the `column` of the control structure keyword (if/else/foreach etc) and ensure the brace is aligned to the same column. This would ensure the brace is always directly under the keyword.

Then you can let the scope indent sniff (or anything else you use) ensure the control structure keyword is aligned properly.
